actors/builtin/multisig/multisig_actor.go

```diff
@@ -344,21 +365,25 @@ func (a Actor) SwapSigner(rt vmr.Runtime, params *SwapSignerParams) *adt.EmptyVa
 
 	var st State
 	rt.State().Transaction(&st, func() {
-		if !isSigner(rt.ResolveAddress, &st, params.From) {
-			rt.Abortf(exitcode.ErrNotFound, "%s is not a signer", params.From)
+		fromIsSigner, fromResolved, err := isSigner(params.From, st.Signers, rt.ResolveAddress, rt.Send)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve to ID address")
+		if !fromIsSigner {
+			rt.Abortf(exitcode.ErrForbidden, "%s is not a signer", params.From)
 		}
 
-		if isSigner(rt.ResolveAddress, &st, params.To) {
+		toIsSigner, toResolved, err := isSigner(params.To, st.Signers, rt.ResolveAddress, rt.Send)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve to ID address")
+		if toIsSigner {
 			rt.Abortf(exitcode.ErrIllegalArgument, "%s already a signer", params.To)
 		}
 
 		newSigners := make([]addr.Address, 0, len(st.Signers))
 		for _, s := range st.Signers {
-			if !isAddressEqual(rt.ResolveAddress, s, params.From) {
+			if s != fromResolved {
 				newSigners = append(newSigners, s)
 			}
 		}
-		newSigners = append(newSigners, params.To)
+		newSigners = append(newSigners, toResolved)
 		st.Signers = newSigners
```

Comment:
Same comment here. I dont think we need to iterate over `st.Signers` anymore.

Reply:
Same comment as above.

Reply:
I guess my question is - can you just do `st.Signers = append(st.Signers, toResolved)`?

Reply:
Iterating over happens twice in this method - once in isSigner, and once here. We only need to iterate a single time, I think!

Reply:
Nevermind, this doesn't work for SwapSigner. But I think it does work for AddSigner!